### PR TITLE
Removed non-critical dependencies from requirements

### LINKED ISSUE #29: Clean requirements
Clean up requirements (e.g. `sphinx` is only needed to build the docs, it is not necessary to run frankenstein).

### PR DESCRIPTION
**Pull request by rbooth200**
This Fixes #29

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -15,6 +15,10 @@
     description="Frankenstein desc",
     long_description=open('README.rst').read(),
     install_requires=[line.rstrip() for line in open("requirements.txt", "r").readlines()],
+    extras_require={
+        'test' : [ 'pytest' ],
+        'docs-build' : [ 'sphinx', 'sphinxcontrib-fulltoc' ],
```
Comment (mtazzari) [inline setup.py]:
```suggestion
        'docs-build' : ['sphinx', 'sphinxcontrib-fulltoc'],
```

File: setup.py [hunk 2]
```diff
@@ -15,6 +15,10 @@
     description="Frankenstein desc",
     long_description=open('README.rst').read(),
     install_requires=[line.rstrip() for line in open("requirements.txt", "r").readlines()],
+    extras_require={
+        'test' : [ 'pytest' ],
```
Comment (mtazzari) [inline setup.py]: Remove spaces

```suggestion
        'test' : ['pytest'],
```

**Comment by rbooth200** [comment]
I've accepted those changes. One thing I wasn't sure about. Could this mess up with for save/restore for the environment on circle CI?

**Comment by mtazzari** [comment]
Good point. I think the circle CI will become slightly slower since the
dependencies cache won't contain the packages listed in the extras
parameter of setup.py.
It must be possible to have different caches for different circle CI jobs
(build/test/deploy), but let's not spend time on this, it's not mission
critical for us.
Moreover, it makes sense to cache the core dependencies.
Happy to merge this.

**Comment by mtazzari** [comment]
Merge whenever you want.

**Pull request merged**